dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassDumperTest.java

```diff
@@ -313,4 +314,18 @@ public void testIsUnusedClassSymbolReference_classSymbolReferenceNotFound()
                   + " constant pool.");
     }
   }
+
+  @Test
+  public void testLoadJavaClass_classNotInRoot()
+      throws IOException, URISyntaxException, ClassNotFoundException {
+    // This JAR file contains com.google.firestore.v1beta1.FirestoreGrpc under BOOT-INF/classes.
+    // Spring-cloud-gcp's spring-cloud-gcp-kotlin-app-sample module generates such file.
+    Path path = absolutePathOfResource("testdata/dummy-boot-inf-prefix.jar");
+    ClassDumper classDumper = ClassDumper.create(ImmutableList.of(path));
+
+    classDumper.findSymbolReferences(); // This builds specialClassFileLocation
+
+    // This should not raise ClassNotFoundException
+    classDumper.loadJavaClass("com.google.firestore.v1beta1.FirestoreGrpc");
```

Comment:
You could assert various tinings about the class returned here.

Reply:
Added assertions